src/backend/src/lib.rs

```diff
@@ -436,6 +439,22 @@ fn list_custom_tokens() -> Vec<CustomToken> {
     read_state(|s| s.custom_token.get(&stored_principal).unwrap_or_default().0)
 }
 
+async fn bitcoin_ecdsa_public_key_of(canister_id: Principal, network: BitcoinNetwork) -> Vec<u8> {
+    let name = read_config(|s| s.ecdsa_key_name.clone());
+    let derivation_path = network_to_derivation_path(network)?;
+    let (key, ) = ecdsa_public_key(EcdsaPublicKeyArgument {
+        canister_id: Some(canister_id),
+        derivation_path,
+        key_id: EcdsaKeyId {
+            curve: EcdsaCurve::Secp256k1,
+            name,
+        },
+    })
+        .await
+        .expect("failed to get public key");
```

Comment:
Given that we are going to have two distinct public keys, the error message can probably be more distinctive as well